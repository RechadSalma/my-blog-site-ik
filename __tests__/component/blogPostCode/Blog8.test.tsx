import { render } from "@testing-library/react";
import Blog8Codes from "../../../components/blogPostCodes/Blog8";

describe("<Blog8/> code for blog8 component", () => {
  const blogEightComponent = () => render(<Blog8Codes />);

  test("renders the <code> tag", () => {
    const { getByRole } = blogEightComponent();

    const codeTag = getByRole("code");

    expect(codeTag).toBeInTheDocument();
  });

  // snapshot component
  test("snapshot Blog8 code tag", () => {
    const { container } = blogEightComponent();

    expect(container).toMatchSnapshot();
  });
});
