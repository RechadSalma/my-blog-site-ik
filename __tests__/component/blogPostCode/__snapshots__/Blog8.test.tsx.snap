// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Blog8/> code for blog8 component snapshot Blog8 code tag 1`] = `
<div>
  <pre
    class="preCode mx-auto"
  >
    <code
      class="codeTag mx-auto"
      role="code"
    >
      <div
        class="comment"
      >
        // Most useful HTML entities
      </div>
      <div>
        &#47; = / (Forward slash)
      </div>
      <div>
        &#33; = ! (Exclamation mark)
      </div>
      <div>
        &#40; = ( (Open parenthesis)
      </div>
      <div>
        &#41; = ) (Close parenthesis)
      </div>
      <div>
        &#44; = , (Comman)
      </div>
      <div>
        &#58; = : (Colon)
      </div>
      <div>
        &#59; = ; (Semi-colon)
      </div>
      <div>
        &#63; = ? (Question mark)
      </div>
      <div>
        &#91; = [ (Open bracket)
      </div>
      <div>
        &#93; = ] (Close bracket)
      </div>
      <div>
        &#92; = \\ (Back slash)
      </div>
      <div>
        &#95; = _ (underscore)
      </div>
      <div>
        &#123; = { (Open curly brace)
      </div>
      <div>
        &#125; = } (Close curly brace)
      </div>
      <div>
        &#124; = | (Pipe)
      </div>
      <div>
        &#160; =   (Empty space)
      </div>
      <div>
        &#8208; = ‐ (Hyphen)
      </div>
      <div>
        &#34; = " (Double quote)
      </div>
      <div>
        &#39; = ' (Single quote)
      </div>
      <div>
        &#96; = \` (Grave accent)
      </div>
      <div>
        &#43; = + (Plus)
      </div>
      <div>
        &#8722; = − (Minus)
      </div>
      <div>
        &#61; = = (Equal)
      </div>
      <div>
        &#60; = &lt; (Less than)
      </div>
      <div>
        &#62; = &gt; (Greater than)
      </div>
      <div>
        &#37; = % (Percentage)
      </div>
      <div>
        &#8727; = ∗ (Asterisk)
      </div>
    </code>
  </pre>
</div>
`;
